tools/IoTEdgeDevOps/VstsPipelineSync/Program.cs

```diff
@@ -2,37 +2,64 @@ namespace VstsPipelineSync
+    using Microsoft.Azure.Services.AppAuthentication;
+    using Microsoft.Azure.KeyVault;
 
     class Program
     {
         static async Task Main(string[] args)
         {
-            (string pat, string dbConnectionString, TimeSpan waitPeriodBeforeNextUpdate, HashSet<string> branches) = GetInputsFromArgs(args);
+            (HashSet<string> branches, TimeSpan waitPeriodBeforeNextUpdate, string pat, string dbConnectionString) = GetInputsFromArgs(args);
             Console.WriteLine($"Wait period before next update=[{waitPeriodBeforeNextUpdate}]");
             await new VstsBuildBatchUpdate(new DevOpsAccessSetting(pat), dbConnectionString, branches).RunAsync(waitPeriodBeforeNextUpdate, CancellationToken.None);
         }
 
-        private static (string pat, string dbConnectionString, TimeSpan waitPeriodBeforeNextUpdate, HashSet<string> branches) GetInputsFromArgs(string[] args)
+        private static (HashSet<string> branches, TimeSpan waitPeriodBeforeNextUpdate, string pat, string dbConnectionString) GetInputsFromArgs(string[] args)
         {
-            if (args.Length != 3 && args.Length != 4)
+            if (args.Length != 2 && args.Length != 4)
             {
-                Console.WriteLine("*** Please provide only 4 parameters - VSTS personal access token, test dashboard database connection string, wait period before next update and branches (with comman delimiter).");
-                Console.WriteLine("Master branch will be automatically included.");
-                Console.WriteLine("VstsBuildBatchUpdate.exe <VSTA PAT> <test dashboard connection string> <wait period, e.g. 00:01:00>");
+                Console.WriteLine("*** This program will ingest vsts data and upload to the database used by the iotedge test dashboard.");
+                Console.WriteLine("By default, it will authenticate with the database and vsts using secrets from keyvault. You can also handle the auth yourself using command line args.");
+                Console.WriteLine("VstsBuildBatchUpdate.exe <branches> <wait-period> [<vsts-pat> <db-connection-string>] ");
+                Console.WriteLine("Usage:");
+                Console.WriteLine(" branches: comma deliminated name of branches");
+                Console.WriteLine(" wait-period: time between db updates (e.g. 00:01:00)");
+                Console.WriteLine(" vsts-pat: personal access token to vsts");
+                Console.WriteLine(" db-connection-string: connection string found in the azure portal");
                 Environment.Exit(1);
             }
 
-            var branches = new HashSet<string> { "refs/heads/master" };
+            HashSet<string> branches = new HashSet<string>(args[0].Split(","));
+            TimeSpan waitPeriodBeforeNextUpdate = TimeSpan.Parse(args[1]);
+            string pat;
+            string dbConnectionString;
             if (args.Length == 4)
             {
-                branches.UnionWith(args[3].Split(","));
+                pat = args[2];
```

Comment:
suggestion: return here directly.

Reply:
I feel like it is cleaner to return once at the end and have a more readable return statement.